### ISSUE TITLE
Stuck saving

### ISSUE THREAD

**Issue by L1Q** [tmux-plugins/tmux-resurrect#379]
Was going to reboot computer, doing the usual resurrect "save" shortcut.
It's stuck on saving.
The rotating animation is going for over 5 minutes, usually under 2 seconds.
Whole tmux and anything inside is completely unusable.
killed the session (or so I though) with ``tmux kill-session``, ``tmux a`` says "no sessions".
starting new tmux session with ``tmux`` (no arguments), it enters same "saving" state that blocks entire tmux session again.
went to ``~/.tmux/resurrect/`` and removed all the files from there
starting new session still presents with saving that is stuck, no new files or symlinks are generated in ``~/.tmux/resurrect/``

What is it doing, how to debug this?
Where else does resurrect keep the state?

First time this is happening after a month or two of using resurrect.

**Comment by L1Q** [comment]
Rebooting entire computer helped, could restore older session.
last session file in ``~/.tmux/resurrect/`` was empty (0 bytes).
new save is successful again.

**Comment by L1Q** [comment]
note: when I said I removed all the files in ``~/.tmux/resurrect/`` I also backed them up so that's how I know the file was empty.

**Comment by LoLei** [comment]
Started happening to me two on the same day on two different machines.

I don't even get the "Saving" notification, but activating the save shortcut simply freezes the session. Attaching to it in another terminal works, the original kept frozen while the newly attached one is unfrozen.

The save file is not being created.

Interestingly enough, activating `prefix + I`, which I used originally to to fetch the plugin and source it via [tpm](https://github.com/tmux-plugins/tpm), produces the same "stuck" behavior.

Running tpm's `install_plugins.sh` script as described here (https://github.com/tmux-plugins/tpm/issues/193#issuecomment-775598298) also does not work, it is just stuck. This (https://github.com/tmux-plugins/tpm/issues/195) does not fix it either. I assume this is the the same thing that's run when the `prefix + I` shortcut is triggered.

So I suspect that this is a bug or incompatibility with tmux itself, as at least for me it lines up with 3.2-1 being released yesterday. (https://archlinux.org/packages/community/x86_64/tmux/ `2021-04-13 08:36 UTC`)

I couldn't read through all of the changes yet (https://github.com/tmux/tmux/issues/2645) but maybe someone else can find something.

**Comment by zhuoqun-chen** [comment]
> Started happening to me two on the same day on two different machines.
> 
> I don't even get the "Saving" notification, but activating the save shortcut simply freezes the session. Attaching to it in another terminal works, the original kept frozen while the newly attached one is unfrozen.
> 
> The save file is not being created.
> 
> Interestingly enough, activating `prefix + I`, which I used originally to to fetch the plugin and source it via [tpm](https://github.com/tmux-plugins/tpm), produces the same "stuck" behavior.
> 
> Running tpm's `install_plugins.sh` script as described here ([tmux-plugins/tpm#193 (comment)](https://github.com/tmux-plugins/tpm/issues/193#issuecomment-775598298)) also does not work, it is just stuck. This ([tmux-plugins/tpm#195](https://github.com/tmux-plugins/tpm/issues/195)) does not fix it either. I assume this is the the same thing that's run when the `prefix + I` shortcut is triggered.
> 
> So I suspect that this is a bug or incompatibility with tmux itself, as at least for me it lines up with 3.2-1 being released yesterday. (https://archlinux.org/packages/community/x86_64/tmux/ `2021-04-13 08:36 UTC`)
> 
> I couldn't read through all of the changes yet ([tmux/tmux#2645](https://github.com/tmux/tmux/issues/2645)) but maybe someone else can find something.
> 
> EDIT: Rebooting does fix it as well. Not sure if permanently though.

yes, rebooting fixs this, otherwise tmux will keep showing `Restoring...` forever after I upgrade to `3.2` yesterday.

**Comment by vkpdeveloper** [comment]
Is there a fix for this or I have to do this everytime, because it's happening a lot in my mac mini.

**Comment by victorwk93812** [comment]
Was working with `/usr/bin/tmux`  on a remote server, and triggered a stuck first on tpm installing plugins, so I killed the SSH session. After a re-login tmux worked with expected theme installed, and I got this exact saving loop bug. Neither deleting files under /tmp nor ~/.local/share/tmux worked. As a result I tried installed the current fresh release under ~/.local/bin. Initially I got a brand new tmux without this problem, so I did the config all over again. Little do I know I've just managed to nuke my new local binary though, and again neither deleting files work nor I see any process under ps. Now both the global and my user binaries behave exactly the same, getting a rotating stick along with "Saving..." forever. Here's my config file 

```
setw -g mode-keys vi
set -g mouse on
set -g default-terminal "screen-256color"
set -ga terminal-overrides ",*256col*:Tc"

# Sets window naming
set -g allow-rename on
set-option -g automatic-rename on
set-option -g automatic-rename-format '#{pane_current_path}'
set-option -g status-interval 1

set -g @continuum-restore 'on'
set -g @continuum-save-interval 3

set -g @catppuccin_flavor "mocha"
set -g @catppuccin_window_status_style "rounded"

# List of plugins
set -g @plugin 'tmux-plugins/tpm'
set -g @plugin 'tmux-plugins/tmux-sensible'
set -g @plugin 'tmux-plugins/tmux-yank'
set -g @plugin 'tmux-plugins/tmux-resurrect'
set -g @plugin 'tmux-plugins/tmux-continuum'
set -g @plugin 'tmux-plugins/tmux-cpu'
set -g @plugin 'tmux-plugins/tmux-battery'
set -g @plugin 'tmux-plugins/tmux-logging'
set -g @plugin 'catppuccin/tmux#latest' # See https://github.com/catppuccin/tmux/tags for additional tags

# Load catppuccin
run ~/.tmux/plugins/tmux/catppuccin.tmux
# Make the status line pretty and add some modules
set -g status-right-length 100
set -g status-left-length 100
set -g status-left ""
set -g status-right "#{E:@catppuccin_status_application}"
set -agF status-right "#{E:@catppuccin_status_cpu}"
set -ag status-right "#{E:@catppuccin_status_session}"
set -ag status-right "#{E:@catppuccin_status_uptime}"
set -agF status-right "#{E:@catppuccin_status_battery}"

# Initialize TMUX plugin manager (keep this line at the very bottom of tmux.conf)
run '~/.tmux/plugins/tpm/tpm'
bind-key -T copy-mode MouseDragEnd1Pane send-keys -X copy-pipe-no-clear pbcopy
bind-key -T copy-mode-vi MouseDragEnd1Pane send-keys -X copy-pipe-no-clear pbcopy
```

fix this pls I have nearly no reboot option.
